### ISSUE TITLE
Things we should deprecate for consistency with Base

### ISSUE THREAD

**Issue by ararslan** [JuliaStats/DataArrays.jl#239]
In the interest of having an API consistent with that of Base, I think we should deprecate the following methods:

- [ ] `DataArray(T, ...)` in favor of `DataArray{T}(...)` (likewise for `PooledDataArray`)
- [ ] Elementwise `isna(X)` in favor of dot-broadcast `isna.(X)`
- [ ] `NAtype` in favor of `NAType` (upper camel case preferred for type names)
- [ ] Elementwise `allna(x)` -> `all(isna, x)`
- [ ] Elementwise `anyna(x)` -> `any(isna, x)`
- [ ] `padNA` -> `padna`, mostly for self-consistency

These functions don't really seem to belong here:

- [ ] `xtab` and `xtabs` -- functionality provided by FreqTables
- [ ] `reldiff` and `percent_change` -- not specific to `DataArray`s and apparently untested

Functions that could use more descriptive names:

- [ ] `gl` (may not belong here though a case could be made)

Thoughts?

**Comment by nalimilan** [comment]
+1, but I would say our priority should be to get DataFrames working on 0.6 first. Else there's little point in having a clean API that cannot be used. :-)

**Comment by ararslan** [comment]
@nalimilan Definitely. These are just things I came across while doing #238 and I figured I'd note them somewhere so that I don't forget. 🙂
